### PR TITLE
docs: address version management recommendation in the readme

### PR DESCRIPTION
**Pull request by KazuCocoa**
To prevent unexpected changes that could break this client's functionality.
Actually, the last break was by a patch version, but maybe it is reasonable to set the limit in the minor version for now.

### CODE REVIEW

**Comment by mykola-mokhnach** [comment]
is it chore or ci?

**Comment by KazuCocoa** [comment]
Removed the ci change from this pr

**Comment by mykola-mokhnach** [comment]
in java we just set the minimum version but don't limit the maximum one 

in general, it has been already confirmed that selenium does not follow semver, so they may introduce breaking changes even in minor releases. For us this means we must either strictly limit the lib version and then manually release a new version of python client when a new selenium is releases and tested for the compatibility. Or just acknowledge that we are limited in resources and we cannot really cover the testing for each version change..

Based on the above statement my proposal would be to follow the same strategy we have in java:
- the selenium version is set to a range where the minimum one is the recent tested/fixed one and the maximum one is the next major
- we allow selenium client updates without further validations. If a compatibility issue happens we fix it, bump the miniumum supported version (if necessary) and release the new version of the python client
- users can always lock/override selenium's version in their projects as a possible workaround

File: setup.py [hunk 1]
```diff
@@ -48,5 +50,5 @@
         'Topic :: Software Development :: Quality Assurance',
         'Topic :: Software Development :: Testing',
     ],
-    install_requires=['selenium ~= 4.1'],
+    install_requires=['selenium >= 4.1, < 4.10'],
```
Comment (mykola-mokhnach) [inline setup.py:53]: not sure why this version should be set here. I though Pipfile is the only source of truth

Comment (KazuCocoa) [reply setup.py:53]: Hm, Pipfile is only in this repository, I thought it was for dev purpose. The package does not include it

Comment (mykola-mokhnach) [reply setup.py:53]: Yes, you are right. Sorry for the confusion: https://gist.github.com/dpboard/887b74f242605c3a409b90f0cf706531

Although I am still not quite convinced about version locking strategy

**Comment by KazuCocoa** [comment]
Correct. Hm, actually maybe what we could as our best is to give users to keep local files in their eng. Let me modify this PR to add acknowledgment in the readme

File: setup.py (hunk 1 above)
Comment (KazuCocoa) [reply setup.py:53]: (sorry, my another comment was reflected slowly)

**Comment by KazuCocoa** [comment]
UPdated

File: setup.py [hunk 2]
```diff
@@ -38,6 +38,8 @@
         'Programming Language :: Python :: 3.7',
         'Programming Language :: Python :: 3.8',
         'Programming Language :: Python :: 3.9',
+        'Programming Language :: Python :: 3.10',
+        'Programming Language :: Python :: 3.11',
```
Comment (mykola-mokhnach) [inline setup.py:42]: shall we change the `install_requires` limit to `< 5.0`?

File: README.md [hunk 3]
```diff
@@ -84,6 +84,13 @@ download and unarchive the source tarball (Appium-Python-Client-X.X.tar.gz).
     python setup.py install
     ```
 
+> **Note**
+> The Appium Python Client depends on [Selenium Python binding](https://pypi.org/project/selenium/), thus
+> the Selenium Python binding update might affect the Appium Python Client.
+> For exampple, some changes in the Selenium binding could break the Appium client.
+> To reduce such unexpected behavior, we strongly recommend you to manage dependencies
+> with version management tool such as Pipenv and Pipfile.
```
Comment (mykola-mokhnach) [inline README.md:92]: Pipfile is just  a config file. The tool itself is called Pipenv.

File: README.md [hunk 4]
```diff
@@ -84,6 +84,13 @@ download and unarchive the source tarball (Appium-Python-Client-X.X.tar.gz).
     python setup.py install
     ```
 
+> **Note**
+> The Appium Python Client depends on [Selenium Python binding](https://pypi.org/project/selenium/), thus
+> the Selenium Python binding update might affect the Appium Python Client.
+> For exampple, some changes in the Selenium binding could break the Appium client.
+> To reduce such unexpected behavior, we strongly recommend you to manage dependencies
```
Comment (mykola-mokhnach) [inline README.md:91]: We probably need to explicitly point out the source for a compatible selenium version. Perhaps, maintaining a table like https://github.com/appium/java-client#compatibility-matrix would also make sense

File: setup.py (hunk 2 above)
Comment (KazuCocoa) [reply setup.py:42]: Existing `selenium ~= 4.1` limits the version up to 4.x, so it is not necessary

File: README.md (hunk 4 above)
Comment (KazuCocoa) [reply README.md:91]: Sounds good. need to recall past versions, so let me prepare as another pr

Comment (KazuCocoa) [reply README.md:91]: -> https://github.com/appium/python-client/pull/876
The version mapping is simpler than Java client's dependencies, so it currently only has a matrix table.

Comment (KazuCocoa) [reply README.md]: Moved into this pr

File: README.md [hunk 5]
```diff
@@ -84,6 +84,23 @@ download and unarchive the source tarball (Appium-Python-Client-X.X.tar.gz).
     python setup.py install
     ```
 
+## Compatibility Matrix
+
+|Appium Python Client| Selenium binding|
+|----|----|
+|`2.10.0`+ |`4.1.0`+ |
```
Comment (mykola-mokhnach) [inline README.md:91]: should it be 4.9.1+ ?

Comment (KazuCocoa) [reply README.md:91]: it worked with older versions, so currently this section does not limit the version. If we need to limit the selenium version, the setup.py will also need to limit it

**Pull request merged**